Src/FluentAssertions/Specialized/ActionAssertions.cs

```diff
@@ -114,6 +114,54 @@ public void NotThrow(string because = "", params object[] becauseArgs)
             }
         }
 
+        /// <summary>
+        /// Asserts that the current <see cref="Action"/> stops throwing any exception
+        /// after a specified amount of time.
+        /// </summary>
+        /// <remarks>
+        /// The <see cref="Action"/> is invoked. If it raises an exception,
+        /// the invocation is repeated until it either stops raising any exceptions
+        /// or the specified wait time is exceeded.
+        /// </remarks>
+        /// <param name="waitTime">
+        /// The time in milliseconds after which the <see cref="Action"/> should have stopped throwing any exception.
+        /// </param>
+        /// <param name="pollInterval">
+        /// The time between subsequent invocations of the <see cref="Action"/>.
+        /// </param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <see cref="because" />.
+        /// </param>
+        /// <exception cref="ArgumentOutOfRangeException">Throws if waitTime or pollInterval are negative.</exception>
+        public void NotThrowAfter(int waitTime, int pollInterval, string because = "", params object[] becauseArgs)
+        {
+            FailIfSubjectIsAsyncVoid();
+
+            if(waitTime < 0 || pollInterval < 0)
+                throw new ArgumentOutOfRangeException("The values of waitTime and pollInterval must be positive.");
+
+            var watch = Stopwatch.StartNew();
+            long invocationEndTime = -1;
+            Exception exception = null;
+
+            while (invocationEndTime < waitTime)
+            {
+                exception = InvokeSubjectWithInterception();
+                if (exception is null)
+                    return;
+
+                Task.Delay(pollInterval).Wait();
```

Comment:
❌ Since this code is synchronous, I would just use `Thread.Sleep`here. Calling `Wait`on a `Task` can be very dangerous.